js/packages/teams-ai/src/authentication/BotAuthentication.ts

```diff
@@ -0,0 +1,280 @@
+/* eslint-disable security/detect-object-injection */
+import {
+    DialogSet,
+    DialogState,
+    DialogTurnResult,
+    DialogTurnStatus,
+    OAuthPrompt,
+    OAuthPromptSettings
+} from 'botbuilder-dialogs';
+import { DefaultTurnState } from '../DefaultTurnStateManager';
+import { TurnState } from '../TurnState';
+import { Application } from '../Application';
+import {
+    ActivityTypes,
+    MemoryStorage,
+    TeamsSSOTokenExchangeMiddleware,
+    TurnContext,
+    Storage,
+    verifyStateOperationName,
+    tokenExchangeOperationName
+} from 'botbuilder';
+import { TurnStateProperty } from '../TurnStateProperty';
+import { AuthError } from './Authentication';
+
+/**
+ * @internal
+ */
+interface OAuthPromptResult {
```

Comment:
Minor: there's a TokenResponse data model from `botbuilder`, and OAuthPrompt returns it as result. May reuse this data model instead of creating our own data model?

```
import {TokenResponse} from 'botbuilder';
```

Reply:
great catch. fixed.